### PR TITLE
Ret 2180

### PR DESCRIPTION
**Pull request by robin-ieong**
https://tools.hmcts.net/jira/browse/RET-2180

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ x] No
```

### CODE REVIEW

File: definitions/json/AuthorisationCaseField.json [hunk 1]
```diff
@@ -15791,6 +15791,546 @@
...
+    "CaseTypeId": "ET_EnglandWales",
+    "CaseFieldID": "hearingAndReferralDetails",
+    "UserRole": "caseworker-employment-legalrep-solicitor",
+    "CRUD": "CRU"
+  },
+  {
+    "CaseTypeId": "ET_EnglandWales",
+    "CaseFieldID": "hearingAndReferralDetails",
+    "UserRole": "caseworker-employment",
+    "CRUD": "CRU"
```
Comment (hsjhita) [inline definitions/json/AuthorisationCaseField.json:15804]: caseworker-employment permissions should always be R

File: definitions/json/CaseField.json [hunk 2]
```diff
@@ -3094,7 +3094,7 @@
   },
   {
     "CaseTypeID": "ET_EnglandWales",
-    "ID": "selectReferralToReply",
+    "ID": "selectReferral",
```
Comment (hsjhita) [inline definitions/json/CaseField.json:3097]: As I said for the data model, shouldn't be changing IDs but it's 'ok' given this isn't live yet

File: definitions/json/CaseEventToFields.json [hunk 3]
```diff
@@ -5451,13 +5451,192 @@
   {
     "CaseTypeID": "ET_EnglandWales",
     "CaseEventID": "replyToReferral",
-    "CaseFieldID": "selectReferralToReply",
+    "CaseFieldID": "selectReferral",
     "DisplayContext": "MANDATORY",
     "PageID": 1,
     "PageDisplayOrder": 1,
     "PageFieldDisplayOrder": 1,
+    "CallBackURLMidEvent": "${ET_COS_URL}/replyReferral/initHearingAndReferralDetails",
```
Comment (JackReevies) [inline definitions/json/CaseEventToFields.json:5459]: url needs to be changed in line with comments in other PR

File: definitions/json/CaseEventToFields.json [hunk 4]
```diff
@@ -5451,13 +5451,192 @@
...
+  },
+  {
+    "CaseTypeID": "ET_EnglandWales",
+    "CaseEventID": "closeReferral",
+    "CaseFieldID": "selectReferral",
+    "DisplayContext": "MANDATORY",
+    "PageID": 1,
+    "PageDisplayOrder": 1,
+    "PageFieldDisplayOrder": 1,
+    "CallBackURLMidEvent": "${ET_COS_URL}/closeReferral/initHearingAndReferralDetails",
```
Comment (JackReevies) [inline definitions/json/CaseEventToFields.json:5594]: url needs to be changed in line with comments in other PR

File: definitions/json/CaseField.json [hunk 5]
```diff
@@ -3106,6 +3106,118 @@
...
+    "HintText": "Give details of your reply to the referral and any issues you've identified",
+    "FieldType": "TextArea",
+    "SecurityClassification": "Public"
+  },
+  {
+    "CaseTypeID": "ET_EnglandWales",
+    "ID": "replyDocument",
+    "Label": "Upload document",
+    "FieldType": "Collection",
+    "FieldTypeParameter": "ReplyToReferralDocumentUpload",
```
Comment (JackReevies) [inline definitions/json/CaseField.json:3158]: Any reason these are their own type and not using DocumentUpload type?

File: definitions/json/ComplexTypes.json [hunk 6]
```diff
@@ -5412,6 +5483,14 @@
...
+  {
+    "ID": "referralDetails",
+    "ListElementCode": "referralReplyCollection",
+    "FieldType": "Collection",
+    "ElementLabel": "Reply",
+    "FieldTypeParameter": "referralReply",
+    "SecurityClassification": "Public"
+  },
   {
     "ID": "CreateReferralDocumentUpload",
```
Comment (JackReevies) [inline definitions/json/ComplexTypes.json:5495]: I know this isnt related to this PR but same as the other comment, why is this its own ComplexType and not just using DocumentUpload?

**Review by JackReevies — CHANGES_REQUESTED** [review]
see comments

File: definitions/json/AuthorisationCaseField.json [hunk 7]
```diff
@@ -15791,6 +15791,546 @@
...
+    "CaseTypeId": "ET_EnglandWales",
+    "CaseFieldID": "directionDetails",
+    "UserRole": "caseworker-employment-legalrep-solicitor",
+    "CRUD": "CRU"
+  },
+  {
+    "CaseTypeId": "ET_EnglandWales",
+    "CaseFieldID": "directionDetails",
+    "UserRole": "caseworker-employment",
+    "CRUD": "R
```
Comment (JackReevies) [inline definitions/json/AuthorisationCaseField.json:15984]: Needs an end quote, this breaks the whole file

File: definitions/json/CaseField.json (hunk 5 above)
Comment (robin-ieong) [reply definitions/json/CaseField.json:3158]: This field doesn't have the document type section: 
image

File: definitions/json/ComplexTypes.json (hunk 6 above)
Comment (robin-ieong) [reply definitions/json/ComplexTypes.json:5495]: I will change it for both of the Create and Reply referral event to use ReferralDocumentUpload type, the complex type can be shared.

File: definitions/json/CaseField.json (hunk 5 above)
Comment (hsjhita) [reply definitions/json/CaseField.json:3158]: But using EventToComplexTypes, you can hide 'Document Type' field

File: definitions/json/ComplexTypes.json (hunk 6 above)
Comment (JackReevies) [reply definitions/json/ComplexTypes.json:5495]: why do we need a new type for this? We can re-use DocumentUpload and just disable that extra field in EventToComplexTypes

Comment (robin-ieong) [reply definitions/json/ComplexTypes.json]: Yep I have made the changes as suggested by Harpreet.

**Review by JackReevies — DISMISSED** [review]

File: definitions/json/ComplexTypes.json [hunk 8]
```diff
@@ -5320,6 +5320,77 @@
...
+    "FieldType": "Text",
+    "ElementLabel": "Directions",
+    "SecurityClassification": "Public"
+  },
+  {
+    "ID": "referralReply",
+    "ListElementCode": "replyDocument",
+    "FieldType": "Collection",
+    "ElementLabel": "Documents",
+    "FieldTypeParameter": "replyDocumentCollection",
```
Comment (hsjhita) [inline definitions/json/ComplexTypes.json:5384]: Thinking this could also be potentially replaced with DocumentUpload?

Comment (robin-ieong) [reply definitions/json/ComplexTypes.json]: Yep, replaced.

**Review by JackReevies — APPROVED** [review]

**Review by hsjhita — APPROVED** [review]

**Pull request merged**